p11-kit 0.23.11 make check fail using homebrew

When trying to install `p11-kit` 0.23.11 using brew I have the following error:

```
==> ./configure --disable-silent-rules --disable-trust-module --prefix=/Volumes/diskE/Users/kakawait/.homebrew/Cellar/p11-kit/0.23.11 --sysconfdir=/Volumes/diskE/Users/kakawait/.homebrew/etc --with-module-config=/Volumes/diskE/Users/kakawait/.homebrew/etc/pkcs11/modules --without-libtas
==> make
==> make check
Last 15 lines from /Volumes/diskE/Users/kakawait/Library/Logs/Homebrew/p11-kit/03.make:
# PASS:  515
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
Please report to https://github.com/p11-glue/p11-kit/issues
============================================================================
make[4]: *** [test-suite.log] Error 1
make[3]: *** [check-TESTS] Error 2
make[2]: *** [check-am] Error 2
make[1]: *** [check-recursive] Error 1
make: *** [check] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting
```

You can find more information inside attached file 
[make.txt](https://github.com/p11-glue/p11-kit/files/2052904/make.txt)

I'm not totally sure if it's more _Homebrew_ or _p11-kit_ issue, feel free to close it if is not directly related to _p11-kit_ itself

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,6 +24,7 @@ matrix:
       sudo: required
       services: docker
       env: BUILD_OPTS="--host=x86_64-w64-mingw32 --without-libffi" CHECK_OPTS="LOG_COMPILER=/srcdir/build/wine-wrapper.sh" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="mingw64-gcc mingw64-libffi mingw64-libtasn1 wine"
+    - os: osx
 
 language: c
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,54 +1,58 @@
-sudo: required
+matrix:
+  include:
+    - os: linux
+      sudo: required
+      services: docker
+      env: BUILD_OPTS="" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS=""
+    - os: linux
+      sudo: required
+      services: docker
+      env: BUILD_OPTS="--enable-coverage" COVERAGE=yes SRCDIR=/coverage BUILDDIR=/coverage EXTRA_PKGS="lcov"
+    - os: linux
+      sudo: required
+      services: docker
+      env: BUILD_OPTS="CFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' CXXFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' LDFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' LIBS='-lasan -ldl -lpthread'" SRCDIR=/srcdir BUILDDIR=/builddir PRELOAD_CMD='ls -1 /usr/lib64/libasan.so.* | head -1' LSAN_OPTIONS='suppressions=/srcdir/build/lsan.supp' EXTRA_PKGS="libasan"
+    - os: linux
+      sudo: required
+      services: docker
+      env: BUILD_OPTS="CFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' CXXFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' LDFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' LIBS='-lubsan -ldl -lpthread'" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="libubsan"
+    - os: linux
+      sudo: required
+      services: docker
+      env: BUILD_OPTS="" SCAN_BUILD="scan-build --status-bugs" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="clang-analyzer"
+    - os: linux
+      sudo: required
+      services: docker
+      env: BUILD_OPTS="--host=x86_64-w64-mingw32 --without-libffi" CHECK_OPTS="LOG_COMPILER=/srcdir/build/wine-wrapper.sh" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="mingw64-gcc mingw64-libffi mingw64-libtasn1 wine"
 
 language: c
 
-env:
-  - BUILD_OPTS="" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS=""
-  - BUILD_OPTS="--enable-coverage" COVERAGE=yes SRCDIR=/coverage BUILDDIR=/coverage EXTRA_PKGS="lcov"
-  - BUILD_OPTS="CFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' CXXFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' LDFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' LIBS='-lasan -ldl -lpthread'" SRCDIR=/srcdir BUILDDIR=/builddir PRELOAD_CMD='ls -1 /usr/lib64/libasan.so.* | head -1' LSAN_OPTIONS='suppressions=/srcdir/build/lsan.supp' EXTRA_PKGS="libasan"
-  - BUILD_OPTS="CFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' CXXFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' LDFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' LIBS='-lubsan -ldl -lpthread'" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="libubsan"
-  - BUILD_OPTS="" SCAN_BUILD="scan-build --status-bugs" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="clang-analyzer"
-  - BUILD_OPTS="--host=x86_64-w64-mingw32 --without-libffi" CHECK_OPTS="LOG_COMPILER=/srcdir/build/wine-wrapper.sh" SRCDIR=/srcdir BUILDDIR=/builddir EXTRA_PKGS="mingw64-gcc mingw64-libffi mingw64-libtasn1 wine"
-
-services:
-  - docker
-
 before_install:
-  - docker pull fedora
-  - export CONTAINER=$(docker run -d fedora sleep 1800)
-  - docker exec $CONTAINER dnf -y install 'dnf-command(builddep)'
-  - docker exec $CONTAINER dnf -y builddep p11-kit
-  - docker exec $CONTAINER dnf -y install gettext-devel git libtool make opensc openssl $EXTRA_PKGS
-  - docker exec $CONTAINER useradd user
+  - |
+      if test -e "./.travis/${TRAVIS_OS_NAME}/before_install.sh"; then
+        . "./.travis/${TRAVIS_OS_NAME}/before_install.sh"
+      fi
 
 install:
-  - docker cp . $CONTAINER:/srcdir
-  - docker exec $CONTAINER cp -R /srcdir /coverage
-  - docker exec $CONTAINER mkdir /builddir
-  - docker exec $CONTAINER chown -R user /builddir
-  - docker exec $CONTAINER mkdir /installdir
-  - docker exec $CONTAINER mkdir -p /installdir/etc/pki/ca-trust-source
-  - docker exec $CONTAINER mkdir -p /installdir/share/pki/ca-trust-source
-  - docker exec $CONTAINER chown -R user /installdir
-  # FIXME: This is needed because some files are included in distribution
-  # and need to be generated in $srcdir rather than $builddir
-  - docker exec $CONTAINER chown -R user /srcdir
-  - docker exec $CONTAINER chown -R user /coverage
+  - |
+      if test -e "./.travis/${TRAVIS_OS_NAME}/install.sh"; then
+        . "./.travis/${TRAVIS_OS_NAME}/install.sh"
+      fi
 
 script:
-  - docker exec $CONTAINER sh -c "cd $SRCDIR && NOCONFIGURE=1 ./autogen.sh"
-  - docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && $SRCDIR/configure --enable-strict --prefix=/installdir --libdir=/installdir/lib --sysconfdir=/installdir/etc --with-trust-paths=/installdir/etc/pki/ca-trust-source:/installdir/share/pki/ca-trust-source $BUILD_OPTS"
   - |
-      if test -n "$PRELOAD_CMD"; then
-        P11_KIT_TEST_LD_PRELOAD=$(docker exec $CONTAINER su - user sh -c "$PRELOAD_CMD")
+      if test -e "./.travis/${TRAVIS_OS_NAME}/script.sh"; then
+        . "./.travis/${TRAVIS_OS_NAME}/script.sh"
       fi
-  - docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && $SCAN_BUILD make -j$(nproc) V=1"
-  - docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && P11_KIT_DEBUG=all LSAN_OPTIONS="$LSAN_OPTIONS" P11_KIT_TEST_LD_PRELOAD=\"$P11_KIT_TEST_LD_PRELOAD\" make check -j$(nproc) V=1 $CHECK_OPTS"
-  - docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && make install"
-  - docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && make installcheck"
 
 after_failure:
-  - docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && cat test-suite.log"
+  - |
+      if test -e "./.travis/${TRAVIS_OS_NAME}/after_failure.sh"; then
+        . "./.travis/${TRAVIS_OS_NAME}/after_failure.sh"
+      fi
 
 after_success:
-  - if test x"$COVERAGE" = xyes; then docker exec $CONTAINER pip install cpp-coveralls; docker exec -e TRAVIS_JOB_ID="$TRAVIS_JOB_ID" -e TRAVIS_BRANCH="$TRAVIS_BRANCH" $CONTAINER sh -c "cd $BUILDDIR && coveralls -b $BUILDDIR -E '(^|.*/)(frob|mock|test)-.*|(^|.*/)(virtual-fixed\.c)' --gcov-options '\-lp'"; fi
+  - |
+      if test -e "./.travis/${TRAVIS_OS_NAME}/after_success.sh"; then
+        . "./.travis/${TRAVIS_OS_NAME}/after_success.sh"
+      fi

--- a/.travis/linux/after_failure.sh
+++ b/.travis/linux/after_failure.sh
@@ -1,0 +1,3 @@
+#!/bin/sh
+
+docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && cat test-suite.log"

--- a/.travis/linux/after_success.sh
+++ b/.travis/linux/after_success.sh
@@ -1,0 +1,9 @@
+#!/bin/sh
+
+if test x"$COVERAGE" = xyes; then
+  docker exec $CONTAINER pip install cpp-coveralls
+  docker exec \
+	 -e TRAVIS_JOB_ID="$TRAVIS_JOB_ID" \
+	 -e TRAVIS_BRANCH="$TRAVIS_BRANCH" \
+	 $CONTAINER sh -c "cd $BUILDDIR && coveralls -b $BUILDDIR -E '(^|.*/)(frob|mock|test)-.*|(^|.*/)(virtual-fixed\.c)' --gcov-options '\-lp'"
+fi

--- a/.travis/linux/before_install.sh
+++ b/.travis/linux/before_install.sh
@@ -1,0 +1,9 @@
+#!/bin/sh
+
+docker pull fedora
+export CONTAINER=$(docker run -d fedora sleep 1800)
+
+docker exec $CONTAINER dnf -y install 'dnf-command(builddep)'
+docker exec $CONTAINER dnf -y builddep p11-kit
+docker exec $CONTAINER dnf -y install gettext-devel git libtool make opensc openssl $EXTRA_PKGS
+docker exec $CONTAINER useradd user

--- a/.travis/linux/install.sh
+++ b/.travis/linux/install.sh
@@ -1,0 +1,14 @@
+#!/bin/sh
+
+docker cp . $CONTAINER:/srcdir
+docker exec $CONTAINER cp -R /srcdir /coverage
+docker exec $CONTAINER mkdir /builddir
+docker exec $CONTAINER chown -R user /builddir
+docker exec $CONTAINER mkdir /installdir
+docker exec $CONTAINER mkdir -p /installdir/etc/pki/ca-trust-source
+docker exec $CONTAINER mkdir -p /installdir/share/pki/ca-trust-source
+docker exec $CONTAINER chown -R user /installdir
+# FIXME: This is needed because some files are included in distribution
+# and need to be generated in $srcdir rather than $builddir
+docker exec $CONTAINER chown -R user /srcdir
+docker exec $CONTAINER chown -R user /coverage

--- a/.travis/linux/script.sh
+++ b/.travis/linux/script.sh
@@ -1,0 +1,11 @@
+#!/bin/sh
+
+docker exec $CONTAINER sh -c "cd $SRCDIR && NOCONFIGURE=1 ./autogen.sh"
+docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && $SRCDIR/configure --enable-strict --prefix=/installdir --libdir=/installdir/lib --sysconfdir=/installdir/etc --with-trust-paths=/installdir/etc/pki/ca-trust-source:/installdir/share/pki/ca-trust-source $BUILD_OPTS"
+if test -n "$PRELOAD_CMD"; then
+  P11_KIT_TEST_LD_PRELOAD=$(docker exec $CONTAINER su - user sh -c "$PRELOAD_CMD")
+fi
+docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && $SCAN_BUILD make -j$(nproc) V=1"
+docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && P11_KIT_DEBUG=all LSAN_OPTIONS="$LSAN_OPTIONS" P11_KIT_TEST_LD_PRELOAD=\"$P11_KIT_TEST_LD_PRELOAD\" make check -j$(nproc) V=1 $CHECK_OPTS"
+docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && make install"
+docker exec $CONTAINER su - user sh -c "cd $BUILDDIR && make installcheck"

--- a/.travis/osx/after_failure.sh
+++ b/.travis/osx/after_failure.sh
@@ -1,0 +1,3 @@
+#!/bin/sh
+
+cat test-suite.log

--- a/.travis/osx/before_install.sh
+++ b/.travis/osx/before_install.sh
@@ -1,0 +1,5 @@
+brew update
+brew install libffi
+
+export PATH=${PATH}:/usr/local/opt/gettext/bin
+export PKG_CONFIG_PATH=${PKG_CONFIG_PATH}:/usr/local/opt/libffi/lib/pkgconfig

--- a/.travis/osx/script.sh
+++ b/.travis/osx/script.sh
@@ -1,0 +1,6 @@
+#!/bin/sh
+
+NOCONFIGURE=1 ./autogen.sh
+./configure --prefix=/installdir --libdir=/installdir/lib --sysconfdir=/installdir/etc --with-trust-paths=/installdir/etc/pki/ca-trust-source:/installdir/share/pki/ca-trust-source
+make V=1
+make check V=1

--- a/common/runtime.c
+++ b/common/runtime.c
@@ -39,6 +39,7 @@
 #include "compat.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 #ifdef OS_UNIX
@@ -58,7 +59,7 @@ p11_get_runtime_directory (char **directoryp)
 	char *directory;
 #ifdef OS_UNIX
 	const char * const *bases = _p11_runtime_bases;
-	char prefix[13 + 1 + 20 + 6 + 1];
+	char *prefix;
 	uid_t uid;
 	struct stat sb;
 	struct passwd pwbuf, *pw;
@@ -84,15 +85,14 @@ p11_get_runtime_directory (char **directoryp)
 	uid = getuid ();
 
 	for (i = 0; bases[i] != NULL; i++) {
-		snprintf (prefix, sizeof prefix, "%s/user/%u",
-			  bases[i], (unsigned int) uid);
+		if (asprintf (&prefix, "%s/user/%u",
+			      bases[i], (unsigned int) uid) < 0)
+			return CKR_HOST_MEMORY;
 		if (stat (prefix, &sb) != -1 && S_ISDIR (sb.st_mode)) {
-			directory = strdup (prefix);
-			if (!directory)
-				return CKR_HOST_MEMORY;
-			*directoryp = directory;
+			*directoryp = prefix;
 			return CKR_OK;
 		}
+		free (prefix);
 	}
 #endif
 

--- a/common/test-runtime.c
+++ b/common/test-runtime.c
@@ -86,6 +86,7 @@ test_bases (void)
 		NULL
 	};
 	char *user, *path;
+	CK_RV rv;
 
 	if (asprintf (&user, "%s/user", test.directory) < 0)
 		assert_not_reached ();
@@ -96,15 +97,13 @@ test_bases (void)
 	free (user);
 	if (mkdir (path, 0700) < 0)
 		assert_not_reached ();
-	free (path);
 
 	bases[0] = test.directory;
 	_p11_runtime_bases = bases;
 
 	unsetenv ("XDG_RUNTIME_DIR");
-	p11_get_runtime_directory (&directory);
-	if (asprintf (&path, "%s/user/%d", test.directory, getuid ()) < 0)
-		assert_not_reached ();
+	rv = p11_get_runtime_directory (&directory);
+	assert_num_eq (CKR_OK, rv);
 	assert_str_eq (path, directory);
 	free (path);
 	free (directory);

